### PR TITLE
Update PCA Docs

### PR DESCRIPTION
**Pull request by md-shafiul-alam**

### CODE REVIEW

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 1]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
-This method uses eigenvalue decomposition of the covariance matrix to compute
-the principal components of the datasets. The method relies on the following
-steps:
+Principal components can be computed by both covariance matrix and correlation matrix.
+The choice of covariance matrix or correlation matrix is application dependent. 
+More specifically, if scaling of the features is important for a problem, 
+which is often the case, using the correlation matrix  to compute principal components is more appropriate.
+By default, oneDAL uses correlation matrix to compute the principal components. But it is possible 
+to use the covariance matrix by calling the ``precomputed`` method and feeding a covariance matrix as input
+to the PCA algorithm. To compute the covariance matrix the :ref:`Covariance <alg_covariance>` algorithm can be used.
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:58]:
```suggestion
to the PCA algorithm. To compute the covariance matrix, the :ref:`Covariance <alg_covariance>` algorithm can be used.
```

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 2]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
+
+The eigenvector associated with the :math:`k`-th largest eigenvalue of the covariance (or correlation) matrix 
+is also the :math:`k`-th principal component of the training data set.  Based on this principle,
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:61]:
```suggestion
is also the :math:`k`-th principal component of the training data set.  Based on this,
```

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 3]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
 .. _pca_t_math_svd:
 
 Training method: *SVD*
 ~~~~~~~~~~~~~~~~~~~~~~
 
-This method uses singular value decomposition of the dataset to compute its
-principal components. The method relies on the following steps:
+The singular value decomposition (SVD) is a matrix factorization technique that
+decomposes an observation matrix :math:`X_{n \times p}` into three matrices as :math:`X = U\SigmaV^*`.
+Here, the columns of :math:`U` are the left-singular vectors, the
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:94]: I'd recommend making it a bullet list. 
Here:
* The columns of :math:`U` are the left-singular vectors.
* The columns of :math:`V` are the right-singular vectors.
* :math:`V^*` is the conjugate transpose of the matrix :math:`V`.
* The diagonal entries of :math:`\Sigma` are the singular values (:math:`\sigma`) of :math:`X`.

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 4]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
+columns of  :math:`V` are the right-singular vectors, :math:`V^*` is the 
+conjugate transpose of the matrix :math:`V` and the diagonal entries of :math:`\Sigma` 
+are the singular values (:math:`\sigma`) of :math:`X`. The right-singular vectors are 
+the principal components of :math:`X`. The steps of computing principal components using SVD technique
+are as follows:
+
+#. Mean centering the input data
+#. Decomposing the mean-centered input data to compute the singular values and the singular vectors
+#. Processing(sorting and storing) the results
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:103]:
```suggestion
#. Processing (sorting and storing) the results
```

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 5]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
 
-#. Computation of the singular values and singular vectors
-#. Formation of the matrices storing the results
+First step is to mean center the input data :math:`M = M_{ij}`, where :math:`M_{ij} = X_{ij} - \frac{\sum_{i=1}^n X_{ij}}{n}`.
 
-To compute singular values :math:`\lambda_i` and singular vectors :math:`u_i`
-and :math:`v_i`, the implementer can choose an arbitrary method such as
-[Demmel90]_.
+Singular values :math:`\sigma_k`, left-singular vectors :math:`U_k` and right-singular vectors :math:`V_k` of matrix :math:`M` can be computed with an arbitrary method like the one described in [Demmel90]_.
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:107]:
```suggestion
Singular values :math:`\sigma_k`, left-singular vectors :math:`U_k`, and right-singular vectors :math:`V_k` of matrix :math:`M` can be computed with an arbitrary method as described in [Demmel90]_.
```

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 6]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
 
-The final step is to sort the set of pairs :math:`(\lambda_i, v_i)` in the
-descending order by :math:`\lambda_i` and form the resulting matrix :math:`T =
-(v_{i,1}, \cdots, v_{i,r}), \quad 1 \leq i \leq p`. Additionally, the means and
+The final step is to find a permutation matrix :math:`Q_{p \times p}` such that the diagonal entries of :math:`\SigmaQ`are sorted in a descending order i.e :math:`\sigma_k \geq \sigma_{k+1}, for all k < p assuming n > p`.
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:109]:
```suggestion
The final step is to find a permutation matrix :math:`Q_{p \times p}` such that the diagonal entries of :math:`\SigmaQ`are sorted in a descending order i.e. :math:`\sigma_k \geq \sigma_{k+1}, for all k < p assuming n > p`.
```

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 7]
```diff
@@ -33,68 +33,89 @@ Mathematical formulation
...
+The rows of the resulting matrix :math:`T = V^*Q` are the principal components of :math:`X`. Note that the rows
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:110]: .. note: The rows of :math:`T` are also the eigenvectors of the covariance matrix of :math:`X`.

File: docs/source/onedal/algorithms/decomposition/pca.rst [hunk 8]
```diff
@@ -33,68 +33,92 @@ Mathematical formulation
...
 
 Sign-flip technique
 ~~~~~~~~~~~~~~~~~~~
-Eigenvectors computed by some eigenvalue solvers are not uniquely defined due to
-sign ambiguity. To get the deterministic result, a sign-flip technique should be
-applied. One of the sign-flip techniques proposed in [Bro07]_ requires the
-following modification of matrix :math:`T`:
+The eigenvectors (or the right-singular vectors) are not uniquely defined because the negative of any eigenvector is also an eigenvector of the input matrix. The signs of the eigenvectors or the singular vectors
+often depend on the solver used. A sign-flip technique like the one proposed in [Bro07]_ helps remove the ambiguity.
+The sign-flip function modifies the matrix :math:`T` as follows:
```
Comment (alexandraepan) [inline docs/source/onedal/algorithms/decomposition/pca.rst:121]:
```suggestion
The sign-flip function modifies the matrix :math:`T` the following way:
```

File: docs/source/includes/decomposition/pca-examples.rst [hunk 9]
```diff
@@ -1,5 +1,5 @@
 .. ******************************************************************************
-.. * Copyright 2020 Intel Corporation
+.. * Copyright 2023 Intel Corporation
```
Comment (KulikovNikita) [inline docs/source/includes/decomposition/pca-examples.rst:2]: I believe this date should reflect the year of this particular file creation

**Review by KulikovNikita — APPROVED** [review]

**Pull request merged**